cli/azd/internal/telemetry/tracer.go

```diff
@@ -98,6 +98,7 @@ type wrapperSpan struct {
 // is called. Therefore, updates to the Span are not allowed after this
 // method has been called.
 func (s *wrapperSpan) End(options ...trace.SpanEndOption) {
+	s.span.SetAttributes(GetGlobalAttributes()...)
```

Comment:
Just wondering - is there a reason we call `SetAttributes` here at the end of the span instead of doing it in `Start` when we propagate the baggage from the context as attributes?  Is it because we end up computing some of these attributes after the span has started?

Reply:
For baggage context propagation, the user has to first set baggage in-context, then create a span with that context, thus we can assert that the baggage properties are created.

For global attributes, there is no synchronization between when an attribute is added and the span being created. Thus, the best thing to do is to add it at the end of the span